### PR TITLE
feat(argo-events): Add annotations to argo-events-sa.

### PR DESCRIPTION
**Pull request by kostas-theo**
This is a small feature to allow the passing in of Kubernetes Service Account (KSA) annotations into the `argo-events-sa` service account. Currently this is not possible using this chart. This is desired to be able to use the already created KSA with GKE workload identity, as well as any other use cases people may have using annotations.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-events/templates/argo-events-sa.yaml [hunk 1]
```diff
@@ -5,7 +5,11 @@ kind: ServiceAccount
 metadata:
   name: {{ .Values.serviceAccount }}
   namespace: {{ .Release.Namespace }}
+  {{- if .Values.serviceAccountAnnotations }}
+  annotations: {{- toYaml .Values.serviceAccountAnnotations | nindent 4 }}
```
Comment (oliverbaehler) [inline charts/argo-events/templates/argo-events-sa.yaml:9]:
```suggestion
  {{- with .Values.serviceAccountAnnotations }}
  annotations: {{- toYaml . | nindent 4 }}
```

File: charts/argo-events/templates/argo-events-sa.yaml [hunk 2]
```diff
@@ -14,5 +18,8 @@ kind: ServiceAccount
 metadata:
   name: {{ $sa }}
   namespace: {{ $namespace }}
+  {{- if $annotations }}
+  annotations: {{- toYaml $annotations | nindent 4 }}
```
Comment (oliverbaehler) [inline charts/argo-events/templates/argo-events-sa.yaml:22]:
```suggestion
  {{- with $annotations }}
  annotations: {{- toYaml . | nindent 4 }}
```

**Review by oliverbaehler — DISMISSED** [review]
You haven't bumped the chart version.

File: charts/argo-events/templates/argo-events-sa.yaml [hunk 3]
```diff
@@ -5,7 +5,11 @@ kind: ServiceAccount
 metadata:
   name: {{ .Values.serviceAccount }}
   namespace: {{ .Release.Namespace }}
+  {{- with .Values.serviceAccountAnnotations }}
+  annotations: {{- toYaml .| nindent 4 }}
```
Comment (mkilchhofer) [inline charts/argo-events/templates/argo-events-sa.yaml:9]:
```suggestion
  annotations: {{- toYaml . | nindent 4 }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Review by mkilchhofer — APPROVED** [review]
LGTM

**Pull request merged**